cogs/pug.py

```diff
@@ -606,18 +693,75 @@ def format_game_server_status(self):
     #########################################################################################
     # Functions:
     #########################################################################################
+    def removeServerReference(self, serverref: str):
+        if serverref in self.current_serverrefs() and serverref not in [None, '']:
+            self.allServers.pop(self.current_serverrefs().index(serverref))
+            return True
+        return False
+
+    def updateServerReference(self, serverref: str, serverdesc: str, serverurl: str = ''):
+        if serverref in self.current_serverrefs() and serverref not in [None, '']:
+            self.allServers.pop(self.current_serverrefs().index(serverref))
+        self.allServers.append((serverref,serverdesc,serverurl))
+        return True
+    
+    def useServer(self, index: int):
```

Comment:
I'd make it require that the index has been adjusted to zero-based in this function, otherwise it gets confusing.

Reply:
Think this is the only comment I was unsure about... Let me know if changes address or not.

Reply:
Yes, where you call this below you're offsetting by -1 so it's correct. I guess you could add something as documentation, but this is fine.